spiceypy/spiceypy.py

```diff
@@ -85,18 +88,67 @@ def spiceFoundExceptionThrower(f):
     @functools.wraps(f)
     def wrapper(*args, **kwargs):
         res = f(*args, **kwargs)
-        found = res[-1]
-        if isinstance(found, bool) and not found:
-            raise stypes.SpiceyError("Spice returns not found for function: {}".format(f.__name__))
-        else:
-            actualres = res[0:-1]
-            if len(actualres) == 1:
-                return actualres[0]
+        if config.catch_false_founds:
+            found = res[-1]
+            if isinstance(found, bool) and not found:
+                raise stypes.SpiceyError("Spice returns not found for function: {}".format(f.__name__))
             else:
-                return actualres
+                actualres = res[0:-1]
+                if len(actualres) == 1:
+                    return actualres[0]
+                else:
+                    return actualres
+        else:
+            return res
     return wrapper
 
 
+@contextmanager
+def no_found_check():
+    """
+    Temporarily disables spiceypy default behavior which raises exceptions for
+    false found flags for certain spice functions. All spice
+    functions executed within the context manager will no longer check the found
+    flag return parameter and the found flag will be included in the return for
+    the given function.
+    For Example bodc2n in spiceypy is normally called like::
+
+        name = spice.bodc2n(399)
+
+    With the possibility that an exception is thrown in the even of a invalid ID::
+
+        name = spice.bodc2n(-999991) # throws a SpiceyError
+
+    With this function however, we can use it as a context manager to do this::
+
+        with spice.no_found_check():
+            name, found = spice.bodc2n(-999991) # found is false, no exception raised!
+
+    Within the context any spice functions called that normally check the found
+    flags will pass through the check without raising an exception if they are false.
+
+    """
+    config.catch_false_founds = False
+    yield
+    config.catch_false_founds = True
```

Comment:
Cool beans!

Consider making this more robust by resetting ```config.catch_false_founds``` to the state it had when no_found_check() was called.  That would protect against code that effectively nests the calls to the context manager.

Say, for example, you write a wrapper for something that uses no_found_check(), but I use it within an app where the first thing I do is call no_found_check() again because I am porting a C program that handles found itself.

These are the changes to make:

in config.py:
```
    config.foundsList = []   # Initialization
```
and then in here:
```
    config.foundsList.append(config.catch_false_founds)
    config.catch_false_founds = False
    yield
    config.catch_false_founds = config.foundsList.pop()
```

Also, the found_check_on/_off() functions below could return the pre-call state, in case someone decided to roll their own context management.

Reply:
good idea, but I think the same effect can be achieved with a local variable, ie:
```
    current_catch_state = config.catch_false_founds
    config.catch_false_founds = False
    yield
    config.catch_false_founds = current_catch_state
```

the on/off calls shouldn't change in the way you suggest as they are not context managers, so there really isn't a before and after for them. It may instead make sense to make another context manager to enable the found catching as a counter point to the one above, and to add a simple method to check to return the current state instead